src/components/Asset/AssetContent/MetaFull.tsx

```diff
@@ -3,17 +3,20 @@ import MetaItem from './MetaItem'
 import styles from './MetaFull.module.css'
 import Publisher from '@shared/Publisher'
 import { useAsset } from '@context/Asset'
-import { useWeb3 } from '@context/Web3'
+// import { useWeb3 } from '@context/Web3'
+import { getDummyWeb3 } from '@utils/web3'
 import { Asset, Datatoken, LoggerInstance } from '@oceanprotocol/lib'
 
 export default function MetaFull({ ddo }: { ddo: Asset }): ReactElement {
   const [paymentCollector, setPaymentCollector] = useState<string>()
   const { isInPurgatory } = useAsset()
-  const { web3 } = useWeb3()
+  // const { web3 } = useWeb3()
 
   useEffect(() => {
     async function getInitialPaymentCollector() {
       try {
+        if (!ddo) return
+        const web3 = await getDummyWeb3(ddo.chainId)
```

Comment:
this means we are always using the fallback web3, even if user has a wallet connected?

Reply:
Is there any disadvantage to using `getDummyWeb3` all the time to request the payment collector address?

Reply:
Not sure from you initial message if using `getDummyWeb3` causes any problems, seemed simple to use that all the time give this is just reading data from chain. Were you suggesting it needs to be changed?

Reply:
I'm not sure if there's any disadvantages to always use ´getDummyWeb3´, but we usually use [this](https://github.com/oceanprotocol/market/blob/8d1782a800d03a194268671b03c39c21446f37ac/src/@utils/fixedRateExchange.ts#L23). Basically we check if there's a web3 connection available; if not, we use ´getDummyWeb3´.